SimonImanigirimpuhwe/my-brand-website-api
# #9 ft(profile): enable updating profile

#### What does this PR do?

- Enable updating profile

#### Description of Task to be completed? 

- Install multer

- Add bio, and photoImage schema properties to the user schema

- Add post endpoint

- Add put request"

- Add get request

- Protect the routes

#### How should this be manually tested?

- Clone the repo 

- In your home directory  cd this branch `ft-update-user-profile-#9`

- Then run `npm install` to install all dependencies

- Run `npm run dev` to start the server

- Use postman send a `PUT` request on `http://localhost:3000/users/profile/` 

> In the request body send `{
        name: name,
       biograph: yourbio
}`

> You should see a returned result with success message and with the updated user info

- Use postman send a `GET` request on `http://localhost:3000/users/profile/` to see all your profile infos

- In the postman send `POST` request `http://localhost:3000/users/profile/` 

> You should see a profile picture saved in the **upload file**

#### Any background context you want to provide?

- To be connected to DB you need to check `env-sample` for the credentials settings

- To be able to update your profile you  need to **login**

- To update profile picture you need to choose in the body  `form-data` and select file

#### What are the relevant Trello stories?

- [#9](https://trello.com/c/NXyVhi56/9-update-user-profile)

#### Screenshots (if appropriate)

<img width="1440" alt="user-profile" src="https://user-images.githubusercontent.com/54619678/96640427-6db08480-1323-11eb-9493-c6931091fa75.png">


#### Questions:

- N/A
